blocks/api/registration.js

```diff
@@ -160,6 +161,14 @@ export function registerBlockType( name, settings ) {
 	if ( ! settings.icon ) {
 		settings.icon = 'block-default';
 	}
+	if ( 'isPrivate' in settings ) {
+		deprecated( 'isPrivate', {
+			version: '3.0',
+			alternative: 'allowedPostTypes',
```

Comment:
`isPrivate` means the block is available programmatically but not in the UI. When I read `allowedPostTypes` or `allowedParents`, it looks like in these case the blocks shouldn't be available entirely (even programmatically). So for me it seems that these two should be separate options? 

The difference is semantic for me, which means if I use isPrivate the block can be inserter everywhere without issue. While we could imagine adding a warning or an error if the block is used at a position that goes against its `allowedParents` config.

Reply:
`isPrivate: true` and `allowedParents: []` behave the same so my thought was to consolidate them. You're right that there's a slight semantic difference, though. I'm happy to add `isPrivate` back 🙂

cc. @gziolo

Reply:
`supports.inserting` kind of makes sense to me. I'm happy with that.